AMW_angular/io/src/app/servers/servers-filter/servers-filter.component.ts

```diff
@@ -0,0 +1,60 @@
+import { ChangeDetectionStrategy, Component, input, output } from '@angular/core';
+import { NgSelectModule } from '@ng-select/ng-select';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { Environment } from '../../deployment/environment';
+import { Resource } from '../../resource/resource';
+import { ServerFilter } from './server-filter';
+import { NgbTypeahead } from '@ng-bootstrap/ng-bootstrap';
+import { Observable, OperatorFunction } from 'rxjs';
+import { debounceTime, distinctUntilChanged, map } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-servers-filter',
+  standalone: true,
+  imports: [NgSelectModule, ReactiveFormsModule, FormsModule, NgbTypeahead],
+  templateUrl: './servers-filter.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ServersFilterComponent {
+  environments = input.required<Environment[]>();
+  runtimes = input.required<Resource[]>();
+  appServerSuggestions = input.required<string[]>();
+  searchFilter = output<ServerFilter>();
+  selectedEnvironmentName: string = 'All';
+  selectedRuntimeName: string = 'All';
+  appServer: string;
+  host: string;
+  node: string;
+
+  search: OperatorFunction<string, readonly string[]> = (text$: Observable<string>) =>
```

Comment:
I have copied and pasted this one (https://ng-bootstrap.github.io/#/components/typeahead/examples), can we have a look at it together? @StephGit

Reply:
I will keep it that way. Thank you!